analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -43,296 +56,278 @@ import org.ossreviewtoolkit.model.Scope
+        val version: String? = null,
 
-    val packages = sortedSetOf<Package>()
-    val issues = mutableListOf<OrtIssue>()
-    val scope = Scope("dependencies", sortedSetOf())
+        val value: String
+    )
 
-    // Maps an (id, version) pair to a (nupkg URL, catalog entry) pair.
-    private val packageReferencesAlreadyFound = mutableMapOf<Pair<String, String>, Pair<String, JsonNode>>()
+    data class Repository(
+        @JacksonXmlProperty(isAttribute = true)
+        val type: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val url: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val branch: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val commit: String? = null
+    )
 
-    init {
-        packageReferences.forEach { id ->
-            val scopeDependency = getPackageReferenceFromRestAPI(id.name, id.version)
-            scopeDependency?.let { scope.dependencies += it }
-        }
+    data class AllPackageData(
+        val data: PackageData,
+        val details: PackageDetails,
+        val spec: PackageSpec
+    )
 
-        scope.collectDependencies().forEach { id ->
-            val pkg = getPackageForId(id)
+    val jsonMapper = JsonMapper().registerKotlinModule()
+    val xmlMapper = XmlMapper(XmlFactory().apply {
+        // Work-around for https://github.com/FasterXML/jackson-module-kotlin/issues/138.
+        xmlTextElementName = "value"
+    }).registerKotlinModule()
 
-            if (pkg != Package.EMPTY) {
-                packages += pkg
-            }
-        }
-    }
+    private val serviceIndex = mapFromUrl<ServiceIndex>(jsonMapper, SERVICE_INDEX_URL)
 
-    private fun getPackageForId(packageId: Identifier) =
-        jsonNodeToPackage(getPackageJsonContent(packageId))
+    // Note: Remove a trailing slash to be able to use manually insert one before the path, as a double-slash would
+    // break the URL!
+    private val registrationsBaseUrl = serviceIndex.resources.single {
+        it.type == REGISTRATIONS_BASE_URL_TYPE
+    }.id.removeSuffix("/")
 
-    private fun getPackageJsonContent(packageId: Identifier): Pair<String, JsonNode> {
-        val (_, _, pkgName, pkgVersion) = packageId
+    private val packageCache = mutableMapOf<Identifier, Pair<AllPackageData, Package>>()
 
-        packageReferencesAlreadyFound[Pair(pkgName, pkgVersion)]?.let {
-            return it
-        }
+    private inline fun <reified T> mapFromUrl(mapper: ObjectMapper, url: String): T {
+        val request = Request.Builder()
+            .get()
+            .url(url)
+            .build()
 
-        return getInformationURL(pkgName, pkgVersion)?.let { (nuspecUrl, catalogUrl) ->
-            Pair(nuspecUrl, jsonMapper.readTree(catalogUrl.requestFromNugetAPI()))
-        } ?: Pair("", EMPTY_JSON_NODE)
-    }
+        val body = OkHttpClientHelper.execute(request) {
+            // Cache responses more aggressively.
```

Comment:
It's not clear to me why it is more aggressive. Would it make sense to explain that in this code comment as well?

Reply:
Done.